model_v4.py

```diff
@@ -0,0 +1,382 @@
+                weights_list[0] = weights_list[0].reshape(src_weights[0].shape)
+                layer.set_weights(weights_list)
+           except (AttributeError, ValueError) as e:
+               print(e)
+
+    def create_conv_pool_layers_set(self, input_layer, addition_name='', conv_filters=64, pool_filters=32, pool_type='max'):
+        """
+        Create 2 additional streams of layers:
+                                            input_layer -> pooling-> conv_1
+                                            input_layer -> conv_2
+        :param input_layer: the last layer
+        :param addition_name: index in the naming system (idx of inception layer)
+        :param conv_filters: num of filters in conv_2 layer
+        :param pool_filters: num of filters in conv_1 layer
+        :param pool_type: 'max' - use MaxPooling or 'avg' - use AveragePooling
+        :return: None
+        """
+        if pool_type == 'avg':
+            mp = AveragePooling2D(pool_size=(3, 3),
+                                  strides=(1, 1),
+                                  padding='same')(input_layer)
+        elif pool_type == 'max':
+            mp = MaxPooling2D(pool_size=(3, 3),
+                              strides=(1, 1),
+                              padding='same')(input_layer)
+        else:
+            raise ValueError('set pool_type max for MaxPooling2D or avg for AveragePooling2D')
+        setattr(self, f'inception_{addition_name}_pool_conv',
+                Conv2D(filters=pool_filters,
+                       kernel_size=(1, 1),
+                       padding='same',
+
+                       kernel_regularizer=l2(0.01),
+                       name=f'inception_{addition_name}_pool_conv')(mp))
+        setattr(self, f'inception_{addition_name}_pool_bn',
+                BatchNormalization(name=f'inception_{addition_name}_pool_bn')(
+                    getattr(self, f'inception_{addition_name}_pool_conv')))
+        setattr(self, f'inception_{addition_name}_pool_act',
+                Activation('relu', name=f'inception_{addition_name}_pool_act')(
+                    getattr(self, f'inception_{addition_name}_pool_bn')))
+
+        setattr(self, f'inception_{addition_name}_1x1_conv',
+                Conv2D(filters=conv_filters,
+                       kernel_size=(1, 1),
+                       padding='same',
+                       kernel_regularizer=l2(0.01),
+                       name=f'inception_{addition_name}_1x1_conv')(input_layer))
+        setattr(self, f'inception_{addition_name}_1x1_bn',
+                BatchNormalization(name=f'inception_{addition_name}_1x1_bn')
+                (getattr(self, f'inception_{addition_name}_1x1_conv')))
+        setattr(self, f'inception_{addition_name}_1x1_act',
+                Activation('relu', name=f'inception_{addition_name}_1x1_act')(
+                    getattr(self, f'inception_{addition_name}_1x1_bn')))
+
+    def create_interception_block(self, input_layer, addition_name='',
```

Comment:
`create_inception_block`, but I propose to split this function into multiple functions, like _inception_block_1a_, _inception_block_1b_. etc, because it significantly improves readability.